proposals/nnnn-audio-io-manager.md

```diff
@@ -0,0 +1,91 @@
+# SDL Audio I/O Manager
+
+* Proposal: [SDL-NNNN](nnnn-audio-io-manager.md)
+* Author: [Kujtim Shala](https://github.com/kshala-ford)
+* Status: **Awaiting review**
+* Impacted Platforms: [ iOS ]
+
+## Introduction
+
+This proposal is about integrating a new manager to the sdl_ios library that allows reading audio data from the car microphone and send audio to the car speaker.
+
+## Motivation
+
+For mobile navigation apps supporting voice based location finding is an important feature allowing address search while driving. In the effort to support SDL partners, Ford worked together with nav app partner to develop a feature for audio input and output. The goal was to use "AudioPassThru" RPCs receiving audio from the car microphone and the audio service session from the SDL protocol specification to play audio.
+
+The feature was originally located in the app as it was a joint development specifically for the app. Due to the substantial use of SDL it required quite some experience with SDL to maintain this code. Moving forward the partner requested to maintain it externally. The feature was refactored as an SDL manager and should now be part of the open source sdl_ios repository. 
+
+## Proposed solution
+
+As described in the motivation, the existing audio input and output feature is implemented in an SDL manager called `SDLAudioIOManager`. The goal is to add this manager to the sdl_ios repository and be part of the official library. Sources are attached to this proposal 
+
+- [SDLAudioIOManager.h](../assets/proposals/nnnn-audio-io-manager/SDLAudioIOManager.h)
+- [SDLAudioIOManager.m](../assets/proposals/nnnn-audio-io-manager/SDLAudioIOManager.m)
+- [SDLAudioIOManagerDelegate.h](../assets/proposals/nnnn-audio-io-manager/SDLAudioIOManagerDelegate.h)
```

Comment:
I have functional code that’s used in production and thought this will avoid requests to the implementation. The proposal is to add this code to the library therefore it should be part of an SDLC proposal.

Reply:
Understood, thanks!